### PR TITLE
Update `chicoma-cpu` modules

### PR DESCRIPTION
**Pull request by altheaden**
This PR updates `chicoma-cpu` to match the changes made in https://github.com/E3SM-Ocean-Discussion/E3SM/pull/112.

Checklist
* [ ] `Testing` comment in the PR documents testing used to verify the changes

### CODE REVIEW

**Comment by altheaden** [comment]
@xylar Regarding our conversation on slack about the path to cmake, I found that `chicoma-cpu_nvidia_mpich.yaml` had a path to cmake already, so I copied that path rather than the one you posted in slack. However, if you would like me to update the path for either or both of these to`/usr/projects/hpcsoft/tce/23-05/cos2-x86_64-cc80/packages/cmake/cmake-3.27.7/`, I can do so.

File: mache/spack/chicoma-cpu_gnu_mpich.csh [hunk 1]
```diff
@@ -44,4 +57,4 @@ setenv PNETCDF_HINTS "romio_ds_write=disable;romio_ds_read=disable;romio_cb_writ
 setenv FI_CXI_RX_MATCH_MODE software
 setenv MPICH_COLL_SYNC MPI_Bcast
 
-setenv LD_LIBRARY_PATH $CRAY_LD_LIBRARY_PATH:$LD_LIBRARY_PATH
+setenv LD_LIBRARY_PATH="/opt/cray/pe/gcc/12.2.0/snos/lib64:${CRAY_LD_LIBRARY_PATH}:${LD_LIBRARY_PATH}"
```
Comment (xylar) [inline mache/spack/chicoma-cpu_gnu_mpich.csh:60]:
```suggestion
setenv LD_LIBRARY_PATH "/opt/cray/pe/gcc/12.2.0/snos/lib64:${CRAY_LD_LIBRARY_PATH}:${LD_LIBRARY_PATH}"
```

**Comment by altheaden** [comment]
@xylar is there anything you would like me to do for this branch/PR or do you have it handled from here?

**Comment by xylar** [comment]
@altheaden, no, no worries for now.  I've totally usurped this branch and I'm testing on Chicoma as I make changes.

File: mache/spack/chicoma-cpu_gnu_mpich.yaml [hunk 2]
```diff
@@ -91,67 +92,55 @@ spack:
...
-        - cray-hdf5-parallel/1.12.2.3
+      - spec: hdf5@1.12.2.9~cxx+fortran+hl~java+mpi+shared
+        prefix: /opt/cray/pe/hdf5-parallel/1.12.2.9/gnu/12.
       buildable: false
     parallel-netcdf:
       externals:
-      - spec: parallel-netcdf@1.12.3.3+cxx+fortran+pic+shared
-        prefix: /opt/cray/pe/parallel-netcdf/1.12.3.3/GNU/9.1/
+      - spec: parallel-netcdf@1.12.3.9+cxx+fortran+pic+shared
+        prefix: /opt/cray/pe/parallel-netcdf/1.12.3.9/gnu/12.3
```
Comment (xylar) [inline mache/spack/chicoma-cpu_gnu_mpich.yaml:126]: For some reason, the module was giving us the path to the nvidia version, not the gnu version.  So I'm trying prefixes again.

**Comment by xylar** [comment]
I think this one is all set.  @altheaden, thanks for getting the ball rolling on this one!

**Pull request merged**